### PR TITLE
test: Make bloom tests deterministic

### PR DESCRIPTION
**Pull request by maflcko**
non-deterministic tests are useless, since a failing test could not be reproduced unless the seed is known.

### CODE REVIEW

File: src/random.h [hunk 1]
```diff
@@ -236,4 +236,8 @@ bool Random_SanityCheck();
  */
 void RandomInit();
 
+/**
+ * Flag to make GetRand return the same number
+ */
+extern bool g_mock_deterministic_tests;
```
Comment (Empact) [inline src/random.h]: nit: Declare `extern` in test code so it's not otherwise accessible in the non-test code?

**Comment by practicalswift** [comment]
Concept ACK

Non-deterministic tests are documented in #15296.

**Comment by laanwj** [comment]
> non-deterministic tests are useless, since a failing test could not be reproduced unless the seed is known

:+1:

File: src/random.h (hunk 1 above)
Comment (MarcoFalke) [reply src/random.h]: Done

**Comment by Empact** [comment]
utACK https://github.com/bitcoin/bitcoin/commit/fae169c95e09ddf068dcaebc8170c4f41b02cf66

**Comment by practicalswift** [comment]
utACK fae169c95e09ddf068dcaebc8170c4f41b02cf66

**Comment by practicalswift** [comment]
Thanks for merging.

Reviewers of this PR might be interested in reviewing #15327 which makes the test `updatecoins_simulation_test` deterministic.

**Pull request merged**